torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -1,14 +1,36 @@
 #include <torch/csrc/distributed/rpc/rpc_agent.h>
+#include <torch/csrc/distributed/autograd/context/dist_autograd_container.h>
+#include <torch/csrc/distributed/autograd/utils.h>
 
 namespace torch {
 namespace distributed {
 namespace rpc {
 
+using namespace torch::distributed::autograd;
+
 RpcAgent::RpcAgent(std::string workerName, RequestCallback cb)
     : workerName_(std::move(workerName)), cb_(std::move(cb)) {}
 
 RpcAgent::~RpcAgent() = default;
 
+std::shared_ptr<FutureMessage> RpcAgent::send(
+    const std::string& to,
+    Message&& message) {
+  // Record appropriate autograd information before sending the message over the
+  // wire.
+  auto& autogradContainer = DistAutogradContainer::getInstance();
+  if (autogradContainer.hasValidContext()) {
```

Comment:
we may not be ready to attach C++ send/recv function for Pyper RPC integration yet.  But init_model_parallel() always init autograd and thus haveValidContext() is true. do we want to pass a disable_dist_auto_grad flag to  init_model_parallel() for now so that Pyper RPC integration can still use Python dist autograd version?

Reply:
`init_model_parallel` doesn't make `hasValidContext()` true. The user needs to explicitly use `with torch.distributed.autograd.context()` around a block of code to enable any of the functionality here.